### PR TITLE
platforms/silicon: remove usage of lib.io.Pin (deprecated).

### LINKED ISSUE #19: `SiliconPlatform` needs to be migrated to Amaranth 0.5
`SiliconPlatform.request()` currently returns an instance of `amaranth.lib.io.Pin`, which is deprecated.

### PR DESCRIPTION
**Pull request by jfng**
Instead, SiliconPlatform.request() returns an instance of SiliconPlatformPort.

Fixes #19.

### CODE REVIEW

**Comment by whitequark** [comment]
I'm confused about this PR. The `SiliconPlatformPort` looks kind of like it's intended to be a `lib.io.PortLike`, but it's not, and doesn't enable using it with `lib.io.Buffer`. Any new cores are going to use `lib.io.Buffer` so this PR isn't sufficient.

**Comment by jfng** [comment]
> I'm confused about this PR. The `SiliconPlatformPort` looks kind of like it's intended to be a `lib.io.PortLike`, but it's not, and doesn't enable using it with `lib.io.Buffer`. Any new cores are going to use `lib.io.Buffer` so this PR isn't sufficient.

It wasn't a `PortLike` because it had no support for `__invert__()`, and even if it was, it wouldn't be usable with `lib.io` buffers without a significant rewrite of the platform.

> If the goal is to just remove the deprecation, `lib.io.Buffer.Signature(...).create()` is sufficient to replace `Pin(...)`.

I updated the PR to return an interface created from `lib.io.Buffer.Signature`.

File: chipflow_lib/platforms/silicon.py [hunk 1]
```diff
@@ -8,7 +8,7 @@
 from amaranth import *
 from amaranth.back import rtlil
 from amaranth.hdl import Fragment
-from amaranth.lib.io import Pin
+from amaranth.lib.io import Direction, Buffer
 from amaranth.hdl._ir import PortDirection
```
Comment (whitequark) [inline chipflow_lib/platforms/silicon.py:12]: I think this private import can now be removed.

**Comment by whitequark** [comment]
> It wasn't a `PortLike` because it had no support for `__invert__()`, and even if it was, it wouldn't be usable with `lib.io` buffers without a significant rewrite of the platform.

Right. I recall that the rewrite of the platform was planned on one of our meetings, which is why I was confused. This PR works fine as a backstop until then.

File: chipflow_lib/platforms/silicon.py (hunk 1 above)
Comment (jfng) [reply chipflow_lib/platforms/silicon.py:12]: Unfortunately not, `PortDirection` is used below (before calling `fragment.prepare()`).

**Review by whitequark — APPROVED** [review]
LGTM, thanks!

**Pull request merged**